### PR TITLE
Package `rust-docs-json` into nightly components (take 3)

### PR DESCRIPTION
**Pull request by jyn514**
`dist` creates a `rust-docs-json.tar.xz` tarfile. But build-manifest expected it to be named `rust-docs-json-preview.tar.xz`. Change build-manifest to allow the name without the `-preview` suffix.

I haven't actually tested this :( build-manifest is a pain to run locally.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
@bors try

Let's get a nightly built from a try build for testing...

**Comment by Mark-Simulacrum** [comment]
Kicked off a dev-static release -- should go through in ~15-30 minutes, and you can test there (`RUSTUP_DIST_SERVER=https://dev-static.rust-lang.org rustup toolchain install nightly`). If that seems to works r=me.

File: src/bootstrap/dist.rs [hunk 1]
```diff
@@ -1365,6 +1365,7 @@ impl Step for Extended {
         }
 
         add_component!("rust-docs" => Docs { host: target });
+        add_component!("rust-json-docs" => JsonDocs { host: target });
```
Comment (LukeMathWalker) [inline src/bootstrap/dist.rs:1368]:
```suggestion
        add_component!("rust-docs-json" => JsonDocs { host: target });
```

**Comment by LukeMathWalker** [comment]
I assume we need another dev-static release to test it out again via `rustup`?

**Comment by Mark-Simulacrum** [comment]
New dev-static release should be ready to go.

**Comment by Mark-Simulacrum** [comment]
Started a dev-static promotion for the try build.

**Comment by jyn514** [comment]
This is finally working :tada: 
```
; RUSTUP_DIST_SERVER=https://dev-static.rust-lang.org rustup component add --toolchain nightly-2022-10-01-x86_64-unknown-linux-gnu rust-docs-json
info: downloading component 'rust-docs-json'
info: installing component 'rust-docs-json
; ls /home/jnelson/.local/lib/rustup/toolchains/nightly-2022-10-01-x86_64-unknown-linux-gnu/share/doc/rust/json
alloc.json  core.json  proc_macro.json  std.json  test.json
```

**Pull request merged**